app/lib/common/providers/space_providers.dart

```diff
@@ -23,6 +23,29 @@ final spacesProvider =
   return SpaceListNotifier(ref: ref, client: client);
 });
 
+/// List of spaces other than current space and it's parent space
+final otherSpacesForInviteMembersProvider = FutureProvider.autoDispose
+    .family<List<Space>, String>((ref, spaceId) async {
+  //GET LIST OF ALL SPACES
+  final allSpaces = ref.watch(spacesProvider);
+
+  //GET PARENT SPACE
+  final parentSpace = ref.watch(canonicalParentProvider(spaceId)).valueOrNull;
+  final parentSpaceId = parentSpace?.space.getRoomIdStr();
+
+  //GET LIST OF SPACES EXCLUDING PARENT SPACE
+  final spacesExcludingParentSpace = allSpaces
```

Comment:
Easier why not loop once and compare against both? Would safe time and memory...

Reply:
True, let me do that